### DISCUSSION TITLE
Support for Rust files

### DISCUSSION THREAD

**Discussion by dukcc** [biomejs/biome#4544] [Ideas]
It would be great for projects and monorepos that use Rust as a back-end to have 1 tool to format and lint all our front-end and back-end code at once. As biome is written in Rust, could this be possible?

**Comment by arendjr** [comment]
While this would be cool, Rust already has a very good official formatter (`rustfmt`, usually invoked through `cargo fmt`) and linter (Clippy, usually invoked through `cargo clippy`), so there's no real reason for Biome to compete with these. We don't intend to act as a frontend for other solutions, so even though this would technically be possible, it's not on our roadmap.

You could consider looking into [`dprint`](https://dprint.dev/) which can act as a universal frontend and has plugins for both Biome and `rustfmt`, but it's for formatting only.
